examples/benchmark.py

```diff
@@ -0,0 +1,311 @@
+            runtimes[key]["mean"] = elapsed.mean()
+            runtimes[key]["std"] = elapsed.std()
+
+        with open(results_file, mode="w") as fd:
+            fd.write("configuration,mean,std,n_iterations,repeat\n")
+            for key, value in runtimes.items():
+                fd.write(f"{key},{value['mean']},{value['std']},{self.n_iter},{self.repeat}\n")
+
+    def log_one_run(self, start, end, model, tag):
+        LOG.info(f"Training Took {end-start} seconds {tag}")
+        LOG.info(f"startprob={model.startprob_}")
+        LOG.info(f"transmat={model.transmat_}")
+
+
+class GaussianBenchmark(Benchmark):
+
+    def new_model(self, implementation):
+        return hmmlearn.hmm.GaussianHMM(
+            n_components=4,
+            n_iter=self.n_iter,
+            covariance_type="full",
+            implementation=implementation,
+            verbose=self.verbose
+        )
+
+    def generate_training_sequences(self):
+        pi = np.asarray([0, 0, 0, 1])
+        A = np.asarray([
+            [.2, .2, .3, .3],
+            [.3, .2, .2, .3],
+            [.2, .3, .3, .2],
+            [.3, .3, .2, .2],
+        ])
+        means = np.asarray([
+            -1.5,
+            0,
+            1.5,
+            3
+        ]).reshape(4, 1)
+        variances = np.asarray([
+            .5,
+            .5,
+            .5,
+            .5
+        ]).reshape(4, 1, 1,)
+
+        sampler = hmmlearn.hmm.GaussianHMM(
+            n_components=4,
+            covariance_type="full",
+            init_params="",
+            verbose=self.verbose
+        )
+        sampler.startprob_ = pi
+        sampler.transmat_ = A
+        sampler.means_ = means
+        sampler.covars_ = variances
+
+        sequences, states = sampler.sample(50_000)
+        lengths = [len(sequences)]
+        return sequences, lengths
+
+    def log_one_run(self, start, end, model, tag):
+        super().log_one_run(start, end, model, tag)
+        LOG.info(f"means={model.means_}")
+        LOG.info(f"covars={model.covars_}")
+
+
+class MultinomialBenchmark(Benchmark):
+
+    def new_model(self, implementation):
+        return hmmlearn.hmm.MultinomialHMM(
+                n_components=3,
+                n_iter=self.n_iter,
+                verbose=self.verbose,
+                implementation=implementation
+            )
+
+    def generate_training_sequences(self):
+        startprob = np.array([0.6, 0.3, 0.1])
+        transmat = np.array([[0.6, 0.2, 0.2],
+                             [0.3, 0.5, 0.2],
+                             [0.4, 0.3, 0.3]])
+
+        emissions = np.array([
+            [.1, .5, .1, .3],
+            [.1, .2, .4, .3],
+            [0, .5, .5, .0],
+        ])
+        sampler = hmmlearn.hmm.MultinomialHMM(n_components=3)
+        sampler.startprob_ = startprob
+        sampler.transmat_ = transmat
+        sampler.emissionprob_ = emissions
+
+        sequences, states = sampler.sample(50_000)
```

Comment:
we still support py35, which doesn't support underscores in literals.

Reply:
edit: actually I'll need to drop py3.5 because there's no manylinux image that supports both py3.5 and py3.10, so no worries there.